### ISSUE TITLE
Two-way data binding not working on TextField

### ISSUE THREAD

**Issue by cafrias** [wwnorton/design-system#28]
Changing the value of the `value` prop won't cause the input field to reflect this new value on its contents.

Here you have a [codesanbox](https://codesandbox.io/s/silent-smoke-y6uii?file=/src/App.js:121-777) to check this live.

```
export default function App() {
  const [pageNumber, setPageNumber] = useState("");

  const handleChange = useCallback(
    (e) => {
      const { value } = e.target;
      setPageNumber(value.slice(0, 6));
    },
    [setPageNumber]
  );

  const setPageNumberTo76 = useCallback(() => {
    setPageNumber("76");
  }, [setPageNumber]);

  return (
    <div className="App">
      <p>Current Page: {pageNumber}</p>
      <TextField value={pageNumber} onChange={handleChange} />
      <button onClick={setPageNumberTo76}>Set page number to 76</button>
    </div>
  );
}
```

In this case when I click on the button, the contents of the input doesn't get updated, as if the input were a Uncontrolled input.

**Comment by sh0ji** [comment]
Gah, that looks like a regression that I introduced in rc.0. [The `value` prop](https://github.com/wwnorton/design-system/blob/b76d19a41af692e7abd573474f768727b7f46651/packages/react/src/components/TextField/index.tsx#L119) isn't being passed to [the `BaseInput`](https://github.com/wwnorton/design-system/blob/b76d19a41af692e7abd573474f768727b7f46651/packages/react/src/components/TextField/index.tsx#L206-L225)! How embarrassing...

Thanks for the report!

**Comment by sh0ji** [comment]
Thanks again for spotting this, @cafrias. I've pushed a fix in #29 and included [a test based on your codesandbox](https://github.com/wwnorton/design-system/pull/29/files#diff-073362a06737690594ddd00d42ade179). When you get a chance, would you mind taking a look at the fix to verify that it resolves your issue before I merge it?

**Comment by cafrias** [comment]
I just checked locally and this solves the problem. Thanks!
